src/components/entity/ha-entity-humidifier.ts

```diff
@@ -0,0 +1,62 @@
+import { css, CSSResult, html, TemplateResult } from "lit-element";
+import { HaEntityToggle } from "./ha-entity-toggle";
+
+class HaEntityHumidifier extends HaEntityToggle {
+  protected render(): TemplateResult {
+    if (!this.stateObj) {
+      return super.render();
+    }
+
+    return html`
+      <div class="target">
+        ${this.stateObj.attributes.mode
+          ? html`<span class="state-label">
+              ${this.hass!.localize(
+                `state_attributes.humidifier.mode.${this.stateObj.attributes.mode}`
+              ) || this.stateObj.attributes.mode}
+            </span>`
+          : ""}
+        <div class="unit">
+          ${this.stateObj.attributes.humidity
+            ? html`${this.stateObj.attributes.humidity} %`
+            : ""}
+        </div>
+      </div>
+
+      ${super.render()}
+    `;
+  }
+
+  static get styles(): CSSResult {
+    return css`
+      :host {
+        display: flex;
+        flex-direction: row;
+        justify-content: center;
+        align-items: center;
+      }
+
+      .target {
+        color: var(--primary-text-color);
+      }
+
+      .current {
+        color: var(--secondary-text-color);
+      }
+
+      .state-label {
+        font-weight: bold;
+        text-transform: capitalize;
+      }
+
+      .unit {
+        display: inline-block;
+        direction: ltr;
+      }
+
+      ${super.styles}
```

Comment:
Make `styles()` return an `CSSResult[]` with your custom css and `super.styles`.

Reply:
Okay, will do that later today

Reply:
Done

Reply:
Seems to be impossible because the base class returns `CSSResult` and not `CSSResult[]`.
Shall I change `HaEntityToggle` to return `CSSResult[]` as well?

Reply:
This should work?
```ts
static get styles(): CSSResult[] {
    return [super.styles, css`.your-styles {}`];
}
```

Reply:
That gives
```
##[error]src/components/entity/ha-entity-humidifier.ts(4,7): error TS2417: Class static side 'typeof HaEntityHumidifier' incorrectly extends base class static side 'typeof HaEntityToggle'.
  Types of property 'styles' are incompatible.
    Type 'CSSResult[]' is missing the following properties from type 'CSSResult': cssText, styleSheet
```

https://github.com/home-assistant/frontend/pull/6083/checks?sha=342f22e6a1f2ab85d0a45dd09681d7f17ac75544

Reply:
And if I change the `HaEntityToggle` class as well, that still gives an error I am trying to understand:
```
##[error]src/components/entity/ha-entity-humidifier.ts(32,7): error TS2739: Type 'CSSResult[]' is missing the following properties from type 'CSSResult': cssText, styleSheet
```
https://github.com/home-assistant/frontend/pull/6083/checks?check_run_id=784665534

Reply:
Here we go!
Found the issue, please check now.

Reply:
That appears to give another issue, truing to understand that now...
```
eval:28:25 TypeError: super.styles.concat is not a function
```

Reply:
Fixed.

Reply:
OK, this isn't better than what you had before :-)

Sorry, can we revert this to your initial code?

(the error you got from typescript is because the entity toggle returns `CSSResult` and you extend that so should return the same type. Changing that to return `CSSResult[]` as well should fix it, but let's not do that)

Reply:
No problem. Done.